client/src/main/java/org/asynchttpclient/netty/channel/DefaultChannelPool.java

```diff
@@ -257,7 +252,7 @@ private boolean offer0(Channel channel, Object partitionKey, long now) {
     }
 
     private void registerChannelCreation(Channel channel, Object partitionKey, long now) {
-        if (channelId2Creation.containsKey(partitionKey)) {
+        if (partitions.containsKey(partitionKey)) {
```

Comment:
Actually, there's a stupid bug here: the test was supposed to be `if (!channelId2Creation.containsKey(partitionKey))`. The goal was to avoid creating a `new ChannelCreation` if the entry is already there. And you don't gain anything here with using `computeIfAbsent` instead of `putIfAbsent` because the lambda would be capturing, and that the key usually exists.

Could you please revert your change and fix the if test?